File: TugasBottomNavigation/app/src/main/java/com/example/tugasbottomnavigation/MainActivity.java
package com.example.tugasbottomnavigation;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity implements BottomNavigationView.OnNavigationItemSelectedListener {

    //Disini saya mengatur supaya semua fragment bisa di klik
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        loadFragment (new com.example.tugasbottomnavigation.HomeFragment());
        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_nav);
        bottomNavigationView.setOnNavigationItemSelectedListener(this);
    }

    private boolean loadFragment(Fragment fragment){
        if (fragment != null){
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.frameContainer, fragment)
                    .commit();
            return true;
        }
        return false;
    }
    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        Fragment fragment = null;
        switch (item.getItemId()){
            case R.id.menu_home:
                fragment = new com.example.tugasbottomnavigation.HomeFragment();
                break;
            case R.id.menu_likes:
                fragment = new com.example.tugasbottomnavigation.LikesFragment();
                break;
            case R.id.menu_uploads:
                fragment = new UploadsFragment();
                break;
            case R.id.menu_search:
                fragment = new SearchFragment();
                break;
            case R.id.menu_profile:
                fragment = new ProfileFragment();
                break;
        }
        return loadFragment(fragment);
    }
}

File: Makanan/app/src/main/java/com/example/makananindonesia/MainActivity.kt
package com.example.makananindonesia

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager

class MainActivity : AppCompatActivity() {
   
    private var list: ArrayList<Foods> = arrayListOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        rvFoods.setHasFixedSize(true)

        list.addAll(FoodsData.listData)
        showRecycler()
    }

    private fun showRecycler() {
        rvFoods.layoutManager = LinearLayoutManager(this)
        val foodAdapter = FoodAdapter(list)
        rvFoods.adapter = foodAdapter
    }
}